pkg/integrations/mssql/common/adapter.go

```diff
@@ -0,0 +1,109 @@
+package common
```

Comment:
I wonder if this whole file would make more sense to have in a separate project (mssql exporter?) or the gathered metrics are tailored for grafana agent needs?

Reply:
I think this is pretty specific to the grafana agent; If someone wanted to run this as a separate process, they'd just use the config we have embedded here with the https://github.com/burningalchemist/sql_exporter.

This code is just for adapting the aforementioned exporter to be compatible with the agent's integration system.

Reply:
nit: should we rename it to collector or something less generic than adapter?